lib/ufe/UsdRotatePivotTranslateUndoableCommand.cpp

```diff
@@ -16,42 +16,40 @@
 
 #include "UsdRotatePivotTranslateUndoableCommand.h"
 #include "private/Utils.h"
+#include "AL/usd/utils/MayaTransformAPI.h"
+#include "../base/debugCodes.h"
 
 MAYAUSD_NS_DEF {
 namespace ufe {
 
-UsdRotatePivotTranslateUndoableCommand::UsdRotatePivotTranslateUndoableCommand(const UsdPrim& prim, const Ufe::Path& ufePath, const Ufe::SceneItem::Ptr& item)
+UsdRotatePivotTranslateUndoableCommand::UsdRotatePivotTranslateUndoableCommand(const UsdPrim& prim, const Ufe::Path& ufePath, const Ufe::SceneItem::Ptr& item, const UsdTimeCode& timeCode)
 	: Ufe::TranslateUndoableCommand(item)
 	, fPrim(prim)
 	, fPath(ufePath)
+	, fTimeCode(timeCode)
 	, fNoPivotOp(false)
 {
-	// Prim does not have a pivot translate attribute
-	const TfToken xpivot("xformOp:translate:pivot");
-	if (!fPrim.HasAttribute(xpivot))
-	{
-		fNoPivotOp = true;
-		// Add an empty pivot translate.
-		rotatePivotTranslateOp(fPrim, fPath, 0, 0, 0);
-	}
-
-	fPivotAttrib = fPrim.GetAttribute(xpivot);
-	fPivotAttrib.Get<GfVec3f>(&fPrevPivotValue);
+	AL::usd::utils::MayaTransformAPI api(prim);
+	fPrevPivotValue = api.rotatePivot(fTimeCode);
 }
 
 UsdRotatePivotTranslateUndoableCommand::~UsdRotatePivotTranslateUndoableCommand()
 {
 }
 
 /*static*/
-UsdRotatePivotTranslateUndoableCommand::Ptr UsdRotatePivotTranslateUndoableCommand::create(const UsdPrim& prim, const Ufe::Path& ufePath, const Ufe::SceneItem::Ptr& item)
+UsdRotatePivotTranslateUndoableCommand::Ptr UsdRotatePivotTranslateUndoableCommand::create(const UsdPrim& prim, const Ufe::Path& ufePath, const Ufe::SceneItem::Ptr& item, const UsdTimeCode& timeCode)
 {
-	return std::make_shared<UsdRotatePivotTranslateUndoableCommand>(prim, ufePath, item);
+	return std::make_shared<UsdRotatePivotTranslateUndoableCommand>(prim, ufePath, item, timeCode);
 }
 
 void UsdRotatePivotTranslateUndoableCommand::undo()
 {
-	fPivotAttrib.Set(fPrevPivotValue);
+	TF_DEBUG(MAYAUSD_UFE_MANIPULATORS).Msg("UsdRotatePivotTranslateUndoableCommand::undo %s (%lf, %lf, %lf) @%lf\n",
+	  fPath.string().c_str(), fPrevPivotValue[0], fPrevPivotValue[1], fPrevPivotValue[2], fTimeCode.GetValue());
+	AL::usd::utils::MayaTransformAPI api(fPrim);
+	api.rotatePivot(GfVec3f(fPrevPivotValue), fTimeCode);
```

Comment:
And so this if where the maya transform api comes into play. I want to provide to TDs a python (or c++) api they can just set a value, with a time, and it just works.

Again, it comes back to business rules. Where does fTimeCode come from?